lib/calendar.ts

```diff
@@ -352,14 +351,15 @@ function resolveNonLunisolarMonth(calendarDate, overflow = undefined, monthsPerY
  * because each object's cache is thrown away when the object is GC-ed.
  */
 class OneObjectCache {
-  constructor(cacheToClone = undefined) {
-    this.map = new Map();
-    this.calls = 0;
+  map = new Map();
+  calls = 0;
+  now: number;
+  hits = 0;
+  misses = 0;
+  constructor(cacheToClone: OneObjectCache = undefined) {
     this.now = globalThis.performance ? globalThis.performance.now() : Date.now();
-    this.hits = 0;
-    this.misses = 0;
     if (cacheToClone !== undefined) {
-      let i = cacheToClone.length;
+      let i = 0; // TODO why was this originally cacheToClone.length ?
```

Comment:
Maybe @justingrant can review this part?

Reply:
Yeah, this looks like a bug. Probably a holdover from a previous merge-based implementation.

Reply:
In that case it should probably get backported right? If I understand the code right, in a hypothetical scenario where one was copying a "full cache" (has MAX_CACHE_ENTRIES things in it), there would be nothing in the new cache?

Reply:
The consequence of this is purely performance, not functional, so unless we're seeing aberrant perf caused by cache exhaustion (AFAIK we're not) then there's no need to back-port to the old polyfill that's just there for running tests. AFAIK there are no tests that fill up the cache, BTW.